tslearn/clustering.py

```diff
@@ -332,6 +342,7 @@ def fit(self, X, y=None, sample_weight=None):
         if n_successful > 0:
             self.X_fit_ = X
             self.labels_ = last_correct_labels
+            self.n_iter_ = n_successful
```

Comment:
Thanks for reviewing @rtavenar! 

This seemed to be a requirement of the check_estimator, perhaps because it inherits from some clustermixin or -base from sklearn? (would have to check) 

I can check if this can be relaxed.

Reply:
OK, but then, it should not be set to `n_successful`, but rather to the number of iterations that were performed for the best run.

Reply:
[sklearn.cluster.SpectralClustering](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.SpectralClustering.html#sklearn.cluster.SpectralClustering) doesn't seem to have a `n_iter_` attribute, yet inherits from `BaseEstimator` and `ClusterMixin`.

However, if we look at the source code for `check_estimator`, we can see that that an estimator that is not a transformer and that has a `max_iter` parameter should have a `n_iter_` attribute:
https://github.com/scikit-learn/scikit-learn/blob/7813f7efb5b2012412888b69e73d76f2df2b50b6/sklearn/utils/estimator_checks.py#L2230-L2232

Reply:
Great, thanks @johannfaouzi! So I think a max_iter hyper-parameter is important for a KMeans algorithm since it is not guaranteed to converge? I will change the value of n_iter_ to be more correct.

Reply:
KMeans is guaranteed to converge to a local minimum. Even for convex problems ([logistic regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) or [support vector machines](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html)), there is a `max_iter` parameter, because the optimization problems must be numerically solved.

I'm not very familiar with `Global Alignment Kernel K-means`, but imho `n_iter_` should only stand for the number of iterations of the optimization process. `GlobalAlignmentKernelKMeans` has a `n_init` parameter, so I don't what `n_iter_` stands for at the moment, but this name should be used for the number of iterations of the optimization process and another name should be used if it has a different meaning.